### PR TITLE
Add KeyFunction support to group in DBCollection

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1979

### CODE REVIEW

**Comment by rozza** [comment]
I plan on backporting to 2.14 after this review is ok'd

File: driver/src/main/com/mongodb/GroupCommand.java [hunk 1]
```diff
@@ -45,12 +48,36 @@
...
+
+    /**
+     * Creates a new group command.
+     *
+     * @param collection the collection from which to perform the group by operation.
+     * @param keyf       the function that creates a "key object" for use as the grouping key
+     * @param condition  optional - a filter to determine which documents in the collection to process.
+     * @param initial    the initial state of the aggregation result document.
+     * @param reduce     a JavaScript aggregation function that operates on the documents during the grouping operation.
+     * @param finalize   optional - a JavaScript function that runs each item in the result set before group returns the final value.
```
Comment (rozza) [inline driver/src/main/com/mongodb/GroupCommand.java]: @Since

File: driver/src/main/com/mongodb/GroupCommand.java [hunk 2]
```diff
@@ -45,12 +48,36 @@
...
+     */
+    public GroupCommand(final DBCollection collection, final String keyf, final DBObject condition,
```
Comment (jyemin) [inline driver/src/main/com/mongodb/GroupCommand.java]: @since

**Comment by jyemin** [comment]
lgtm after @since

**Comment by rozza** [comment]
Updated, rebased and merged

**Pull request closed without merging**